framework/transport/ieee1905_transport/ieee1905_transport_netlink.cpp

```diff
@@ -70,11 +70,15 @@ int Ieee1905Transport::handle_netlink_message(struct nlmsghdr *msghdr)
     if (msghdr->nlmsg_type == RTM_NEWLINK || msghdr->nlmsg_type == RTM_DELLINK) {
         struct ifinfomsg *ifi = (struct ifinfomsg *)NLMSG_DATA(msghdr);
         bool is_active        = (ifi->ifi_flags & IFF_RUNNING) && (ifi->ifi_flags & IFF_UP);
-
+        std::string ifname    = if_index2name(ifi->ifi_index);
```

Comment:
Note that the message is followed by the attributes of the interface, one of which is IFLA_IFNAME. So we could just parse those attributes instead of opening a socket (which if_index2name does).

(nitpick)

Reply:
I think that parsing the attribute here is less readable than using indextoname, and the overhead of creating a socket is negligible.